assets/js/modules/analytics-4/components/dashboard/EnhancedMeasurementActivationBanner/SetupBanner.js

```diff
@@ -0,0 +1,208 @@
+	const [ isSaving, setIsSaving ] = useState( false );
+
+	const hasEditScope = useSelect( ( select ) =>
+		select( CORE_USER ).hasScope( EDIT_SCOPE )
+	);
+
+	const autoSubmit = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' )
+	);
+
+	const documentationURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getGoogleSupportURL( {
+			path: '/analytics/answer/9216061',
+		} )
+	);
+
+	const { submitChanges } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const { setPermissionScopeError } = useDispatch( CORE_USER );
+	const { setValues } = useDispatch( CORE_FORMS );
+
+	const commonSubmitChanges = useCallback( async () => {
+		setIsSaving( true );
+
+		setValues( ENHANCED_MEASUREMENT_FORM, {
+			[ ENHANCED_MEASUREMENT_ENABLED ]: true,
+		} );
+
+		const { error } = await submitChanges();
+
+		setIsSaving( false );
+
+		if ( error ) {
+			setErrorNotice( error );
+			return;
+		}
+
+		// Ask the parent component to show the success banner.
+		// This should be called last because it will unmount this component.
+		onSubmitSuccess();
+	}, [ setValues, submitChanges, onSubmitSuccess ] );
+
+	const handleSubmitChanges = async () => {
+		const scopes = [];
+
+		if ( hasEditScope === false ) {
+			scopes.push( EDIT_SCOPE );
+		}
+
+		// If scope not granted, trigger scope error right away. These are
+		// typically handled automatically based on API responses, but
+		// this particular case has some special handling to improve UX.
+		if ( scopes.length > 0 ) {
+			setValues( FORM_SETUP, { autoSubmit: true } );
+
+			setPermissionScopeError( {
+				code: ERROR_CODE_MISSING_REQUIRED_SCOPE,
+				message: __(
+					'Additional permissions are required to enable enhanced measurement for the selected web data stream.',
+					'google-site-kit'
+				),
+				data: {
+					status: 403,
+					scopes,
+					skipModal: true,
+					redirectURL: global.location.href,
+				},
+			} );
+			return;
+		}
+
+		await commonSubmitChanges();
+	};
+
+	// If the user lands back on this component with autoSubmit and the edit scope,
+	// resubmit the form.
+	useEffect( () => {
+		async function handleAutoSubmit() {
+			// Auto-submit should only auto-invoke once.
+			setValues( FORM_SETUP, { autoSubmit: false } );
+
+			await commonSubmitChanges();
+		}
+
+		if ( autoSubmit && hasEditScope ) {
+			handleAutoSubmit();
+		}
+	}, [
+		hasEditScope,
+		setValues,
+		commonSubmitChanges,
+		onSubmitSuccess,
+		autoSubmit,
+	] );
+
+	const description = hasEditScope
+		? __(
+				'Enable enhanced measurement in Analytics to automatically track metrics like file downloads, video plays, form interactions, etc. No extra code required.',
+				'google-site-kit'
+		  )
+		: __(
+				'Enable enhanced measurement in Analytics to automatically track metrics like file downloads, video plays, form interactions, etc. No extra code required — you’ll be redirected to give permission for Site Kit to enable it on your behalf.',
```

Comment:
This text variant for when the edit scope is not present was added after discussion with Mariya on [Slack](https://10up.slack.com/archives/CFFRMC5DE/p1696002454202519).